File: arena-jface/src/main/java/org/uqbar/lacar/ui/impl/jface/JFaceStyledTextBuilder.java
package org.uqbar.lacar.ui.impl.jface;

import java.util.List;

import org.eclipse.swt.SWT;
import org.scalatest.Entry;
import org.uqbar.arena.widgets.style.Style;
import org.uqbar.lacar.ui.impl.jface.bindings.JFaceBindingBuilder;
import org.uqbar.lacar.ui.model.BindingBuilder;
import org.uqbar.ui.jface.controller.KeyWordText;

/**
 * 
 * @author nnydjesus
 */
public class JFaceStyledTextBuilder extends JFaceSkinnableControlBuilder<KeyWordText> {

	public JFaceStyledTextBuilder(JFaceContainer container, List<Entry<String[], Style>> configuration) {
		super(container, new KeyWordText(container.getJFaceComposite(), configuration));
	}

	@Override
	public BindingBuilder observeValue() {
		return new JFaceBindingBuilder(this, new KeyWordTextObservableValue(this.getWidget(), SWT.Modify));
	}
}

File: arena-jface/src/main/scala/org/uqbar/lacar/ui/impl/jface/builder/JFaceStyledTextBuilder.scala
package org.uqbar.lacar.ui.impl.jface.builder

import org.uqbar.swt.widgets.KeyWordText
import org.uqbar.arena.widgets.style.Style
import org.uqbar.lacar.ui.model.BindingBuilder
import org.uqbar.lacar.ui.impl.jface.bindings.JFaceBindingBuilder
import org.eclipse.swt.SWT
import org.uqbar.lacar.ui.impl.jface.builder.traits.JFaceContainer
import org.uqbar.lacar.ui.impl.jface.bindings.observables.KeyWordTextObservableValue

/**
 * Created by jfernandes on 4/8/14.
 */
class JFaceStyledTextBuilder(container: JFaceContainer, configuration: java.util.Map[Array[String], Style])
  extends JFaceSkinnableControlBuilder[KeyWordText](container, new KeyWordText(container.getJFaceComposite, configuration)) {

  override def observeValue = {
    new JFaceBindingBuilder(this, new KeyWordTextObservableValue(widget, SWT Modify))
  }

}
